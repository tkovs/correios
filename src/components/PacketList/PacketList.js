import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'
import { Divider } from 'react-native-paper'
import isEmpty from 'lodash/isEmpty'

import PacketItem from '../PacketItem'

const PacketList = ({ packets }) => {
  if (isEmpty(packets)) {
    return null
  }

  return packets.map(packet => (
    <View style={styles.container} key={packet.code}>
      <PacketItem packet={packet} />
      <Divider />
    </View>
  ))
}

const styles = StyleSheet.create({
  container: {
    marginLeft: 8,
  },
})

PacketList.propTypes = {
  packets: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default PacketList
