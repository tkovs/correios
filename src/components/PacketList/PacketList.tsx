import React, { FC } from 'react'
import { View, StyleSheet, ScrollView, Dimensions } from 'react-native'
import { Divider, Caption } from 'react-native-paper'
import Icon from 'react-native-vector-icons/MaterialIcons'
import isEmpty from 'lodash/isEmpty'

import PacketItem from '../PacketItem'

interface PacketListProps {
  packets: Packet[]
}

type Props = PacketListProps

const PacketList: FC<Props> = ({ packets }: Props) => {
  const screenWidth = Dimensions.get('window').width
  const iconSize = screenWidth * 0.8
  const iconColor = '#ddd'

  if (isEmpty(packets)) {
    return (
      <View style={styles.iconScreen}>
        <Icon name="error-outline" size={iconSize} color={iconColor} />
        <Caption>Nenhuma encomenda nesta lista</Caption>
      </View>
    )
  }

  return (
    <ScrollView>
      {packets.map((packet) => (
        <View style={styles.container} key={packet.code}>
          <PacketItem packet={packet} />
          <Divider accessibilityStates={['disabled']} />
        </View>
      ))}
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 8,
  },
  iconScreen: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
  },
})

export default PacketList
