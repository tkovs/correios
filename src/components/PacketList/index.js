import { connect } from 'react-redux'

import PacketList from './PacketList'

const mapStateToProps = ({ packets }) => ({
  packets,
})

export default connect(mapStateToProps)(PacketList)
