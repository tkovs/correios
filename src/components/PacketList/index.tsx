import React, { FC } from 'react'
import { connect, MapStateToProps } from 'react-redux'

import PacketList from './PacketList'
import { packetsListWithoutArchivedSelector } from '../../store/selectors/packets'

interface OwnProps {
  filter?: string
}

interface StateProps {
  packets: Packet[]
}

type Props = StateProps & OwnProps

const PacketListContainer: FC<Props> = (props: Props) => {
  const { packets: packetsProp, filter } = props
  const packets = filter
    ? packetsProp.filter((packet) => packet.status === filter)
    : packetsProp

  return <PacketList packets={packets} />
}

const mapStateToProps: MapStateToProps<StateProps, OwnProps, State> = (
  state: State
) => ({
  packets: packetsListWithoutArchivedSelector(state),
})

export default connect(mapStateToProps)(PacketListContainer)
