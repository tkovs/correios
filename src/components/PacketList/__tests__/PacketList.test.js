import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import PacketList from '../PacketList'
import mockPackets from '../../../__mocks__/packets.json'

it('renders correctly with packets', () => {
  const { baseElement } = render(<PacketList packets={mockPackets} />)
  expect(baseElement).toMatchSnapshot()
})

it('renders correctly without packets', () => {
  const { baseElement } = render(<PacketList packets={[]} />)
  expect(baseElement).toMatchSnapshot()
})
