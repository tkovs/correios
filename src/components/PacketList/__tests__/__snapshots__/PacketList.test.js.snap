// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with packets 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView>
      <View
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#E5E5E5"
        >
          <View
            style={
              Object {
                "color": "#f9f9f9",
                "marginVertical": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#141414",
                }
              }
            >
              iPhone 7
            </Text>
            <Text
              style={
                Object {
                  "color": "#3E8989",
                  "fontWeight": "bold",
                }
              }
            >
              delivered
            </Text>
            <Text
              style={
                Object {
                  "color": "#939393",
                }
              }
            >
              PW086958109BR
            </Text>
          </View>
        </TouchableHighlight>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
    </ScrollView>
  </View>
</View>
`;

exports[`renders correctly without packets 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
